consensus/src/quorum_store/proof_manager.rs

```diff
@@ -88,23 +128,137 @@ impl ProofManager {
+                                        // TODO: Can cloning be avoided here?
+                                        if BatchPayload::new(batch.author(), txns.clone()).hash()
+                                            == *batch.digest()
+                                        {
+                                            inline_txns += batch.num_txns();
+                                            inline_bytes += batch.num_bytes();
+                                            cur_txns += batch.num_txns();
+                                            cur_bytes += batch.num_bytes();
+                                            inline_block.push((batch.clone(), txns.clone()));
+                                            inline_batch_count += 1;
+                                        } else {
+                                            counters::NUM_CORRUPT_BATCHES.inc();
+                                        }
+                                    }
+                                    true
+                                } else {
+                                    full = true;
+                                    false
+                                }
+                            } else {
+                                false
+                            }
+                        })
+                    }
+                }
+                counters::NUM_INLINE_BATCHES.observe(inline_batch_count as f64);
+                counters::NUM_INLINE_TXNS.observe(inline_txns as f64);
+                counters::NUM_TXN_SPACE_LEFT_IN_PROPOSAL.observe((max_txns - cur_txns) as f64);
+
                 let res = GetPayloadResponse::GetPayloadResponse(
-                    if proof_block.is_empty() {
+                    if proof_block.is_empty() && inline_block.is_empty() {
                         Payload::empty(true)
-                    } else {
+                    } else if inline_block.is_empty() {
                         trace!(
                             "QS: GetBlockRequest excluded len {}, block len {}",
                             excluded_batches.len(),
                             proof_block.len()
                         );
                         Payload::InQuorumStore(ProofWithData::new(proof_block))
+                    } else {
+                        trace!(
+                            "QS: GetBlockRequest excluded len {}, block len {}, inline len {}",
+                            excluded_batches.len(),
+                            proof_block.len(),
+                            inline_block.len()
+                        );
+                        // TODO: Need to calcuale max_txns_to_execute correctly here.
+                        Payload::QuorumStoreInlineHybrid(
+                            inline_block,
+                            ProofWithDataWithTxnLimit::new(
+                                ProofWithData::new(proof_block),
+                                Some(max_txns as usize),
```

Comment:
This should be `None` and replaced later?

Reply:
Done.